packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -279,20 +279,44 @@ class FuchsiaDevice extends Device {
         printError('Failed to start the Fuchsia package server');
         return LaunchResult.failed();
       }
+
+      // Serve the application's package.
       final File farArchive = package.farArchive(
           debuggingOptions.buildInfo.mode);
       if (!await fuchsiaPackageServer.addPackage(farArchive)) {
         printError('Failed to add package to the package server');
         return LaunchResult.failed();
       }
 
+      // Serve the flutter_runner.
+      final File flutterRunnerArchive = fuchsiaArtifacts.flutterRunnerForTarget(
+        await targetPlatform,
+        buildMode: debuggingOptions.buildInfo.mode,
+      );
+      if (!await fuchsiaPackageServer.addPackage(flutterRunnerArchive)) {
+        printError('Failed to add flutter_runner package to the package server');
+        return LaunchResult.failed();
+      }
+
       // Teach the package controller about the package server.
       if (!await fuchsiaDeviceTools.amberCtl.addRepoCfg(this, fuchsiaPackageServer)) {
         printError('Failed to teach amber about the package server');
         return LaunchResult.failed();
       }
       serverRegistered = true;
 
+      // Tell the package controller to prefetch the flutter_runner.
+      String flutterRunnerName = 'flutter_jit_runner';
+      if (!debuggingOptions.buildInfo.isDebug &&
+          !debuggingOptions.buildInfo.isProfile) {
+        flutterRunnerName = 'flutter_jit_product_runner';
```

Comment:
we should use the aot runner

Reply:
(can be a separate PR of course)

Reply:
Yeah, the tool doesn't know how to build aot Fuchsia packages, yet, so that'll probably be a bigger change.